sdk/scheduler/src/main/java/com/mesosphere/sdk/state/StateStoreUtils.java

```diff
@@ -121,7 +119,59 @@ public static TaskInfo getTaskInfo(StateStore stateStore, TaskStatus taskStatus)
                     "Failed to find a task with TaskID: %s", taskStatus));
         }
 
-        return taskInfoOptional.get();
+        return taskInfoOptional.get().getName();
+    }
+
+    /**
+     * TaskInfo and TaskStatus objects referring to the same Task name are not written atomically.
+     * It is therefore possible for the states across these elements to become out of sync.  While the scheduler process
+     * is up they remain in sync.  This method guarantees produces an initial synchronized state.
+     *
+     * For example:
+     * <ol>
+     * <li>TaskInfo(name=foo, id=1) is written</li>
+     * <li>TaskStatus(name=foo, id=1, status=RUNNING) is written</li>
+     * <li>Task foo is reconfigured/relaunched</li>
+     * <li>TaskInfo(name=foo, id=2) is written</li>
+     * <li>Scheduler is restarted before new TaskStatus is written</li>
+     * <li>Scheduler comes back and sees TaskInfo(name=foo, id=2) and TaskStatus(name=foo, id=1, status=RUNNING)</li>
+     * </ol>
+     *
+     * Note that this ID mismatch would specifically cause problems {@link #getTaskName(StateStore, TaskStatus)} which
+     * requires that the ids align.
+     */
+    static void repairTaskIDs(StateStore stateStore) {
```

Comment:
Does this method guarantee synchronous  access? That is to say, is there another method that may be updating the `stateStore` at the same time as this one being called?